src/client/java/src/main/java/com/vesoft/nebula/graph/client/GraphClientImpl.java

```diff
@@ -0,0 +1,185 @@
+
+            Random random = new Random(System.currentTimeMillis());
+            int index = random.nextInt(nodes.size());
+            InetSocketAddress activeNode = nodes.get(index);
+
+            String activeHost;
+            InetAddress address = activeNode.getAddress();
+            if (address instanceof Inet6Address) {
+                activeHost = InetAddresses.toUriString(address);
+            } else {
+                activeHost = InetAddresses.toAddrString(address);
+            }
+
+            transport = new TSocket(activeHost,
+                    activeNode.getPort(), timeout);
+            TProtocol protocol = new TBinaryProtocol(transport);
+
+            try {
+                transport.open();
+                client = new GraphService.Client(protocol);
+                AuthResponse result = client.authenticate(username, password);
+                if (result.getError_code() == ErrorCode.E_BAD_USERNAME_PASSWORD) {
+                    LOGGER.error("User name or password error");
+                    return ErrorCode.E_BAD_USERNAME_PASSWORD;
+                }
+
+                if (result.getError_code() != ErrorCode.SUCCEEDED) {
+                    LOGGER.error(String.format("Host : %s error : %s",
+                            activeHost, result.getError_msg()));
+                } else {
+                    sessionId_ = result.getSession_id();
+                    return ErrorCode.SUCCEEDED;
+                }
+            } catch (TTransportException tte) {
+                LOGGER.error("Connect failed: " + tte.getMessage());
+            } catch (TException te) {
+                LOGGER.error("Connect failed: " + te.getMessage());
+            }
+        }
+        return ErrorCode.E_FAIL_TO_CONNECT;
+    }
+
+    @Override
+    public void disconnect() {
+        if (!checkTransportOpened(transport)) {
+            return;
+        }
+
+        try {
+            client.signout(sessionId_);
+        } catch (TException e) {
+            LOGGER.error("Disconnect error: " + e.getMessage());
+        } finally {
+            transport.close();
+        }
+    }
+
+    @Override
+    public int execute(String stmt) {
+        if (!checkTransportOpened(transport)) {
+            return ErrorCode.E_DISCONNECTED;
+        }
+
+        try {
+            ExecutionResponse executionResponse = client.execute(sessionId_, stmt);
+            if (executionResponse.getError_code() != ErrorCode.SUCCEEDED) {
+                LOGGER.error("execute error: " + executionResponse.getError_msg());
+            }
+            return executionResponse.getError_code();
+        } catch (TException e) {
+            LOGGER.error("Thrift rpc call failed: " + e.getMessage());
+            return ErrorCode.E_RPC_FAILURE;
+        }
+    }
+
+    @Override
+    public int executeUpdate(String stmt) {
+        return execute(stmt);
+    }
+
+    @Override
+    public ResultSet executeQuery(String stmt) throws NGQLException {
```

Comment:
NGQLException doesn't seem to be enough. In addition to `ResultSet`, `executeQuery` should contain these status values that `execute` method will return.